### DISCUSSION TITLE
How to disable setup page entirely?

### DISCUSSION THREAD

**Discussion by weirdyang** [OrchardCMS/OrchardCore#15467] [Q&A] [answered]
I'm currently using auto-setup, however sometimes on first load the set up page appears, and once I refresh I will be brough to the login screen. How do I disable the setup page entirely?

**Comment by MarGraz** [comment]
Hi,

I have never encountered this issue before. 
It's important to note that the auto-setup feature is based on a `Transient` injected service. This means its lifecycle is unique per request. For a good explanation of different DI lifetimes, check out this [article](https://grantwinney.com/difference-between-singleton-scoped-transient/). 

Additionally, I recommend reviewing the [documentation](https://docs.orchardcore.net/en/latest/docs/reference/modules/AutoSetup/#enabling-auto-setup-feature) on auto-setup, which explains how this feature is enabled in your code.

Hope this helps 😊

**Reply by weirdyang** [reply]
I'm not quite sure how reviewing the documentation helps. I have set up the application as per the documentation.
![image](https://github.com/OrchardCMS/OrchardCore/assets/25576022/d4518d87-1c40-4536-bc2a-16d48669ebf8)
![image](https://github.com/OrchardCMS/OrchardCore/assets/25576022/34f8d3f6-8dcb-4857-a188-4622d9321dc2)
the .env file is also read.

**Reply by MarGraz** [reply]
Hi, the documentation shows you how this feature is enabled, so if you want to deactivate it, you need to remove the "AutoSetup" service from the Startup file.

**Reply by MarGraz** [reply]
Remember also to check your appsettings.json and how you have set the auto setup. In the [documentation](https://docs.orchardcore.net/en/latest/docs/reference/modules/AutoSetup/#json-configuration-parameters) you can find a table that explains the values.

**Reply by weirdyang** [reply]
Please read my original post, there is no issue with auto setup, it is just a bit slow sometimes and shows the set up page. I have no issue with the settings. My question is how do I disable the set up page.

**Reply by MarGraz** [reply]
So you don't want to disable auto setup, but the setup page? 🤔

The way to [bypass the setup page](https://github.com/OrchardCMS/OrchardCore/pull/4567#issue-508960944) is using a recipe. In this way a recipe will publishes your Orchard Core configuration on Startup. 
The auto setup will look for an Orchard Core configuration in appsettings.json or in the environment variables and provision your site.

The Startup module is involved. See also the [documentation](https://docs.orchardcore.net/en/main/docs/reference/modules/Setup/) for some examples.

I hope I get your necessity.

**Comment by weirdyang** [comment, marked as answer]
The current workaround is replace the the Setup module index page in your application.
